types/wordpress__components/wordpress__components-tests.tsx

```diff
@@ -581,8 +581,7 @@ const kbshortcuts = {
             actions: [{ label: "Action One", url: "https://wordpress.org" }],
         },
     ]}
->
-</C.NoticeList>;
+/>;
```

Comment:
This shouldn't need changing. `<div></div>` will not add any `children` to the props so typing this as `children: never[]` is incorrect. But from recent React TS bug reports, I don't think this will get adressed.

@andarist Do you know when this changed? Should've been somewhere between 5.2 and 5.3

Reply:
Could you prepare a playground that would showcase the problem? I just put this into a playground (using TS nightly: [TS playground](https://www.typescriptlang.org/play?ts=5.3.0-dev.20231002#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wG4AoUSWOAKjhQGc4BhObXAgAQHdoATMEUaMA9GlyQAdkikxGZcuQA8LAHQA5CDGBokAGWCMY5OHDQAbJow0oQSALz4QATwC0U7bqRuLRmPimcJ46eowOAN4A2kFmEbFmcMD8AFwEAIz4ADQJZhJysjBp+AASSBYWENm5DBjAEFKMaVERcFYARuXFAIJ1DXAA8jLZcACuUBbFABYwMGBNoqJ8UILCjGrQAOb4cAC+ALo5iXtHcPu75AB8yqLqWqEG-pcUQA)) and I don't see any problem. At a quick glance here I'm just not sure what I should be looking for

Reply:
I don't know what TS version CI is running on but I didn't make the error message up: https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/6381109561/job/17316900896

Will check in two days again to see if this is either brand new or fixed. Though it looks like TS playground and CI have different versions of TS nightly.

Reply:
Both the CI and my TS playground used the same TS version from what I can tell. Maybe the difference is in the tsconfig options? It's quite hard to tell - I still can't repro this in the playground despite trying a bunch of things. I also tried this locally with a very similar `tsconfig.json` to the one that I could find here.

Reply:
Jake rechecked that this isn't a TS problem but rather something might have changed in `@types/react` or smth. The same thing happens with TS 5.2: [TS playground](https://www.typescriptlang.org/play?ts=5.2.2#code/JYWwDg9gTgLgBAbzgYQuCA7AphmAVATzCwBo4BZCAVwGcsBRANxxgAkBDDAEwBssoyAJSzsAxjAByELljgBfOADMoaOACIoI8WoDcAWABQoSLDjCx8Zao1aYuw4Zmie7TXAzsQWGmDGypMMCisgiGcOFwwLj8in5wAAoqYDSIYRHpAPQAVFlp6RFZKJgwLHAwEHCa3PyRGGUAFrIYEIHBAHR5+XBZGZ3povXAPFxVAFxmtlIy+gZd4dm5s3PdcACCda5Q7ARwEIpwFsCYuwBGAFZY4jQdS3M9fRGHmDQA-OOa7FyYPDur4kcYADaAF04AAfOBUaqKKJYLgzZbOdg0GgSTxYN5wGgwKBRADm4Mh0Nh8Ie8xyZPChQA6o0YI0oA0mi0grIaPVqMM4CdZFxgDQQPyaMATnxdozmjAbssVgABGSxKg8GAANXYPCoshxmsp3V6t3y-IAIvzBSiRXxMScIBA+JxCVCFSSEXMFrrCgAxKH-Y6idV8LhwADujTqfIFQvxTPcLPa7v1y0wwhAEGYLwAFABKcaMCDAUkGuSdKIlKCxYJwABCyKwf0Cx1CBvSLh5PHG2NxGDxLq6SJRaK8mI7UYhjqwMOwBeWbqb6UKAGUOUrAwr2Er4H26CkeZUsCnmIGrCBo08MC945Tmkbx2vlcgXCjvFabXa6qPiZOe-Ji9Ey3FK1QMDlBgdYAnAWAAB4lNwKTVnQoENpSmD3kEADW4yULQDDMLgHDcHwUAADysHg5AADIAUBmD0HwXi4AAfF+RYGiWMRxAAqoIZEIXUkHQVwsE1jxqSzuEVBQG2WI4viTGdDARCyMJAC8VaAcBwkQpx3E+hgMzMU4LhuKIzzwAErLjKg6DYLghDEIRZntIkEDJIxDgGAZrhNOiPhxA5WBkfy8CNukrF-hWTnJCJ04UqJVJrFAWw7HsMatN4ZQVFUMhQNKyz3LFkqsjQ7wiF8GA-HAfkgl+mQxTKnrevWdR+jwAbBqGBwpayWJLlyO6aPucJwBkcDhmadBcDldwJnMSZ7qmGLpgVwQAJJcO20ldtmcC5vm1URJuA4YutnYEu+Tqfp0zEhb+5b+LGsh8TgAlwAA8oKMD2fdbQRTQZBqAMQwjDgaiEmos0DWo9FRXM+bHTJlLGdEuDjOY4hTFgskGPplyGbIiPYhV90BdiFloJA1n4Apn2pcTUo-a5BiGNTrK050B3okpaggAQAC0S1YDzPCBWonT8zQSkIIClLBTK4Sw+oACMagkLq+3FCw4xqKwWAtRAyuq+Ep5FXAgJIC2Oua8JL3YMrkISZr9RAckowZBkQbQFwYCaCibTQHiINyMCKuxXIwfpMCzH0UzGR+bTDNM7HgVsw+qIc1zvP84Lwui-d4uS9LBuRGtiv67F-Tq8j6ja7rpey4bOnG6bcDm5JahWzbZDia3jswM7rvu1Anve9cfsB0HlKh50EeGBkUcGEAA)

Reply:
It's odd because these tests ran when we removed implicit children. But then it was always incorrect that `<Component></Component>` adds `children`. Has an easy workaround so I'm not too concerned even though it imposes some stylistic rules.

PS: [reduced repro](https://www.typescriptlang.org/play?jsx=4&ts=5.2.2#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwCwAUI8AHYxZQF5ZwAKxYdAG9GcMXFwAbNKgByyEFgD8ALjioYUVgHM4AHzgBXFgBMsBVlhP0GAX0YFj+YBBZxZEGMFxYAMsA0ACjABVDV+CEEASjgRBnE4ShhDKDcWQ0lJG3smBgAeDy8ffw1RcSkZeUUAXjIQAE8AWhZPbyxGyQCYMkYAPkY8gHpCtpKYXptGRiA)

Reply:
It definitely looks odd. And it acts differently between those two despite the fact that the JS output is the same:
```ts
const errors = <NoticeList
    className="my-notice-list"
>
</NoticeList>;

const ok = <NoticeList
    className="my-notice-list"
></NoticeList>;
```